server/task-workers/shared-memory-manager.cpp

```diff
@@ -0,0 +1,85 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2020 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include <cassert>
+#include <cstddef>
+#include <new>
+#include <random>
+#include <sys/mman.h>
+
+#include "server/task-workers/shared-context.h"
+#include "server/task-workers/shared-memory-manager.h"
+
+namespace task_workers {
+void task_workers::SharedMemoryManager::init() {
+  slice_size = slice_payload_size + slice_meta_info_size;
+  total_slices_count = memory_size / slice_size;
+
+  assert(slice_meta_info_size >= sizeof(SliceMetaInfo));
+  assert(slice_payload_size % 1024 == 0);
+  assert(memory_size % slice_size == 0);
+
+  memory = static_cast<unsigned char *>(mmap(nullptr, memory_size, PROT_READ | PROT_WRITE, MAP_SHARED | MAP_ANONYMOUS, -1, 0));
+  for (size_t i = 0; i < total_slices_count; ++i) {
+    unsigned char *slice_ptr = get_slice_ptr(i);
+    new (slice_ptr + slice_payload_size) SliceMetaInfo{};
+  }
+}
+
+void *SharedMemoryManager::allocate_slice() {
+  std::random_device rd;
+  std::mt19937 gen(rd());
```

Comment:
You need to generate only one random number, so I suggest to use `std::random_device` directly without the mt generator.
Also, please, move `std::random_device` into the class, we don't need to create it on the each call (I'm not sure about the cost of `std::random_device` creation).